Visualization Updates

LBern opened the pull request:
# General
This PR introduces two new viz commands from _alice_: `visualize_xyzi_pointclouds_from_mission` and `visualize_xyzi_pointclouds_sequentially`.
These are nice visualizations functions that allow you to do stuff as in: [Team CERBERUS - Urban Circuit - Beta Course](https://www.youtube.com/watch?v=0-GbFrrWcL4)

ethzasl-jenkins commented:
Test PASSed.

smauq commented on console-plugins/vi-map-visualization-plugin/include/vi-map-visualization-plugin/visualization-plugin.h:
```diff
@@ -31,6 +30,8 @@ class VisualizationPlugin : public common::ConsolePluginBase {
   int visualizeSensorExtrinsics() const;
   int visualizeLandmarkObserverRays() const;
   int visualizeReprojectedDepthResource(backend::ResourceType type);
+  int visualizeReprojectedDepthResourceFromMission(backend::ResourceType type);
+  int visualizeReprojectedDepthResourceSequentially(backend::ResourceType type);
```
Maybe I'm missing something obvious but I don't see where these functions are defined?

LBern replied:
good catch! I forgot to copy something, my bad!

smauq commented on visualization/src/resource-visualization.cc:
```diff
@@ -419,4 +432,160 @@ void visualizeReprojectedDepthResourcePerRobot(
+  for (const vi_map::MissionId& mission_id : mission_ids) {
+    const std::string& mission_string = mission_id.shortHex();
+    VLOG(1) << "mission string is " << mission_string;
+    if (!FLAGS_vis_pointcloud_mission_id.empty() &&
```
I think it would make more sense to me to also here move this mission selection up to the calling function which would either pass what I assume is all mission_ids or a list with a single mission id. Otherwise the function interface is a bit confusing since depending on the flag it might ignore the input arguments.

LBern replied:
in fact, we don't need this flag at all! the visualization plugin already has a flag that can filter missions. -- removed this part.

smauq reviewed (commented):
A few interface changes I think and I'm maybe missing something obvious with the function definitions. Otherwise looks good.

smauq reviewed (changes requested):
Clicked just "comment" by accident

LBern commented on visualization/src/resource-visualization.cc:
```diff
@@ -419,4 +432,160 @@ void visualizeReprojectedDepthResourcePerRobot(
+      integration_function);
+}
+
+void visualizeReprojectedDepthResourceFromMission(
```
it is being called by the plugin (from console):

```
addCommand(
      {"visualize_xyzi_pointclouds_from_mission"},
      [this]() -> int {
        return visualizeReprojectedDepthResourceFromMission(
            backend::ResourceType::kPointCloudXYZI);
      },
      "Publish all xyz + intensity point clouds.", common::Processing::Sync);
```

LBern replied:
which then contains the line:
```
  visualization::visualizeReprojectedDepthResourceFromMission(
```

LBern commented on visualization/src/resource-visualization.cc:
```diff
@@ -63,6 +63,17 @@ DEFINE_bool(
+    "Specifies the mission for publishing the dense map.");
+DEFINE_string(
+    vis_pointcloud_mission_id_topic, "",
+    "Specifies the topic for the mission specific dense map.");
```
i added a default value `/mission_point_cloud`. I'd not put the same topic as it is a different pointcloud and I don't want that someone replaces the current `/point_cloud` by accident with it.

LBern commented on visualization/src/resource-visualization.cc:
```diff
@@ -419,4 +432,160 @@ void visualizeReprojectedDepthResourcePerRobot(
+
+void visualizeReprojectedDepthResourceFromMission(
+    const backend::ResourceType input_resource_type,
+    const vi_map::MissionIdList& mission_ids, const vi_map::VIMap& vi_map) {
```
I changed the interface. the viz plugin can already filter missions and we rely on this now. There we can also return an error if the mission flag hasn't been specified.

LBern commented:
> A few interface changes I think and I'm maybe missing something obvious with the function definitions. Otherwise looks good.

@smauq 
Thanks for the review! I changed the interface for the `visualizeReprojectedDepthResourceFromMission` function and removed the flag. The function `visualizeReprojectedDepthResourceSequentially` now fully relies on whatever the viz plugin gives it.

ethzasl-jenkins commented:
Test PASSed.

smauq reviewed (approved):
LGTM!

ethzasl-jenkins commented:
Test FAILed.

ethzasl-jenkins commented:
Test PASSed.
